src/app/modules/shared_models/member_model.nim

```diff
@@ -190,6 +190,21 @@ QtObject:
     self.endRemoveRows()
     self.countChanged()
 
+  proc removeAllItems(self: Model) =
+    if self.items.len <= 0:
+      return
+
+    let parentModelIndex = newQModelIndex()
+    defer: parentModelIndex.delete
+  
+    let startIdx = 0
+    let endIdx = startIdx + self.items.len - 1
+  
+    self.beginRemoveRows(parentModelIndex, startIdx, endIdx)
```

Comment:
You could as well (preferably) `begin/endResetModel` instead and not bother with the indexes :)